### PR TITLE
#1401. Async for-in element tests added

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Patterns/execution_pattern_for_in_element_A02_t01.dart [hunk 1]
```diff
@@ -26,7 +26,7 @@
 ///   append(<element>);
 /// }
 /// ```
-/// @description Checks that it is a compile-time error if type check ot the
+/// @description Checks that it is a compile-time error if the type check ot the
```
Comment (eernstg) [inline LanguageFeatures/Patterns/execution_pattern_for_in_element_A02_t01.dart:29]:
```suggestion
/// @description Checks that it is a compile-time error if the type check of the
```

File: LanguageFeatures/Patterns/execution_pattern_for_in_element_A02_t02.dart [hunk 2]
```diff
@@ -26,7 +26,7 @@
 ///   append(<element>);
 /// }
 /// ```
-/// @description Checks that it is a compile-time error if type check ot the
+/// @description Checks that it is a compile-time error if the type check ot the
```
Comment (eernstg) [inline LanguageFeatures/Patterns/execution_pattern_for_in_element_A02_t02.dart:29]:
```suggestion
/// @description Checks that it is a compile-time error if the type check of the
```

File: LanguageFeatures/Patterns/execution_pattern_await_for_in_element_A06_t03.dart [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+///   an operation to add an element to the surrounding collection being built:
+/// ```dart
+/// I id1 = <expression>;
+/// Iterator<E> id2 = id1.iterator;
+/// while (id2.moveNext()) {
+///   <keyword> <pattern> = id2.current;
+///   append(<element>);
+/// }
+/// ```
+/// @description Checks that it is a compile-time error if type of an async
```
Comment (eernstg) [inline LanguageFeatures/Patterns/execution_pattern_await_for_in_element_A06_t03.dart:29]:
```suggestion
/// @description Checks that it is a compile-time error if the type of an async
```

**Review by eernstg — COMMENTED** [review]
Looks good! Just noted a handful of typos.

File: LanguageFeatures/Patterns/execution_pattern_await_for_in_element_A03_t01.dart [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    2
+  ];
+  Expect.listEquals([0, 3, 2], l7);
+  Expect.isTrue(Storing.stored is Stream<(num, {num n})>);
+
+  var l8 = [
+    1,
+    await for (final (v9, n: v10)
+        in Stream<(num, {num n})>.fromIterable([(1, n: 2)]))
+      v10.expectStaticType<Exactly<num>>(),
```
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Patterns/execution_pattern_await_for_in_element_A03_t01.dart:117]: OK, it is borderline silly to add that extra case, but we might as well cover the "positional record component" case as well, given that all the setup around it is already in place. ;-)

**Comment by sgrekhov** [comment]
Thanx. Updated. Please take another look

File: LanguageFeatures/Patterns/execution_pattern_await_for_in_element_A03_t01.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Patterns/execution_pattern_await_for_in_element_A03_t01.dart:118]: Thanks!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**